docs/developers/how-to-build-plugins.md

```diff
@@ -100,169 +129,205 @@ This will append the new parameter informatrion to the object loaded from `param
 
-For absolute metrics like carbon, the right value is `sum` because you would want to add carbon emissions from each timestep when you aggregate over time.
+## Appendix: Walk-through of the Sum plugin
 
-For proportional metrics, the right value is `avg`. For example, you would want to calculate the average `cpu-utilization` - it would not make sense to sum it when aggregating over multiple timesteps.
+To demonstrate how to build a plugin that conforms to the `pluginInterface`, let's examine the `sum` plugin.
 
-Finally, values that should always be presented identically regardless of any aggregation, such as names or global constants, should be given the `aggregation-method` value `none`.
+The `sum` plugin manifestements the following logic:
 
-Now you are ready to run your model using the `impact-engine` CLI tool!
+- sum whatever is provided in the `input-parameters` field from `globalConfig`.
+- append the result to each element in the output array with the name provided as `output-parameter` in `globalConfig`.
 
-## Running your model
+Let's look at how you would manifestement this from scratch:
 
-### Linking local model
+The plugin must be a function conforming to `PluginInterface`. You can call the function `Sum`, and inside the body you can add the signature for the `execute` method:
 
-For using locally developed model in `if` please follow these steps:
+```typescript
+export const Sum = (globalConfig: SumConfig): PluginInterface => {
+  const errorBuilder = buildErrorMessage(Sum.name);
+  const metadata = {
+    kind: 'execute',
+  };
 
-1. On the root level of a locally developed model run `npm link`, which will create a global package. It uses `package.json` file's `name` field as a package name. Additionally, name can be checked by running `npm ls -g --depth=0 --link=true`.
+  /**
+   * Calculate the sum of each input.
+   */
+  const execute = async (inputs: PluginParams[]): Promise<PluginParams[]> => {
 
-2. Use the linked model in impl by specifying `name`, `model`, `path` in initialize models section.
+  };
 
-### Using model directly from github
+  return {
+    metadata,
+    execute,
+  };
 
-You can simply save your model in a public Github repository and pass the path to it in your impl.
+}
+```
 
-For example, for a model saved in `github.com/my-repo/my-model` you can do the following:
+Your plugin now has the basic structure required for IF integration. Your next task is to add code to the body of `execute` to enable the actual plugin logic to be manifestemented.
 
-npm install your model:
+The `execute` function should grab the `input-parameters` (the values to sum) from `globalConfig`. it shoudl then iterate over the `inputs` array, get the values for each of the `input-parameters` and append them to the `inputs` array, using the name from the `output-parameter` value in `globalConfig`. Here's what this can look like, with the actual calculation pushed to a separate function, `calculateSum`. 
```

Comment:
```suggestion
The `execute` function should grab the `input-parameters` (the values to sum) from `globalConfig`. it should then iterate over the `inputs` array, get the values for each of the `input-parameters` and append them to the `inputs` array, using the name from the `output-parameter` value in `globalConfig`. Here's what this can look like, with the actual calculation pushed to a separate function, `calculateSum`. 
```